src/tsm/tsm-screen.c

```diff
@@ -283,9 +283,19 @@ static void screen_scroll_up(struct tsm_screen *con, unsigned int num)
 	 * also be small enough so we do not get stack overflows. */
 	if (num > 128) {
 		screen_scroll_up(con, 128);
-		return screen_scroll_up(con, num - 128);
+		screen_scroll_up(con, num - 128);
+		return;
+	}
+
+#ifdef _MSC_VER
+	struct line **cache = malloc(sizeof(struct line *) * num);
+	if (!cache) {
+		llog_warning(con, "Out of memory");
+		return;
 	}
+#else
 	struct line *cache[num];
+#endif
```

Comment:
Using non-constant variables as array size is nonstandard anyway. Let's just use the malloc version on all platforms.